dialect/sql/schema/postgres.go

```diff
@@ -526,3 +526,94 @@ func arrayType(t string) bool {
 	}
 	return true
 }
+
+func (d *Postgres) foreignKeys(ctx context.Context, tx dialect.Tx, tables []*Table) (map[*Table][]*ForeignKey, error) {
+	var fks = make(map[*Table][]*ForeignKey)
+
+	var tableLookup = make(map[string]*Table)
+
+	// TODO: include schema
+	for _, t := range tables {
+		tableLookup[t.Name] = t
+	}
+
+	for _, t := range tables {
+		rows := &sql.Rows{}
+		query := fmt.Sprintf(fkQuery, t.Name)
+		if err := tx.Query(ctx, query, []interface{}{}, rows); err != nil {
+			return nil, fmt.Errorf("querying foreign keys for table %s: %w", t.Name, err)
+		}
+		defer rows.Close()
+		var (
+			tableFksLookup = make(map[string]*ForeignKey)
+			// Maintain stable order
+			tableFks []*ForeignKey
+		)
```

Comment:
```suggestion
		fksLookup := make(map[string]*ForeignKey)
```

Append directly to the table.